Dockerfile

```diff
@@ -1,25 +1,29 @@
-FROM python:3.8-slim AS build
-
-# Set our workdir
+# adapted from https://dev.to/farcellier/package-a-poetry-project-in-a-docker-container-for-production-3b4m
+# and  https://medium.com/@albertazzir/blazing-fast-python-docker-builds-with-poetry-a78a66f5aed0
+FROM amazon/aws-lambda-python:3.12 AS build
 WORKDIR /app
 
 # Get ready to build
-RUN pip install build
+RUN pip install --no-cache-dir poetry==1.8.3
+
+ENV POETRY_NO_INTERACTION=1 \
+    POETRY_VIRTUALENVS_IN_PROJECT=1 \
+    POETRY_VIRTUALENVS_CREATE=1 \
+    POETRY_CACHE_DIR=/tmp/poetry_cache
 
-# Now copy the app over and build a wheel
-COPY src /app/src
-COPY pyproject.toml setup.cfg /app/
-RUN python -m build
+COPY pyproject.toml poetry.lock ./
+RUN touch /app/README.md
+COPY athena_federation ./athena_federation
+RUN poetry build -f wheel
 
-## Now use the compiled wheel in our lambda function
-FROM amazon/aws-lambda-python:3.8.2021.12.09.15 AS lambda
+FROM amazon/aws-lambda-python:3.12 AS lambda
+WORKDIR /app
 
-ENV TARGET_BUCKET=replace_me
+ENV TARGET_BUCKET=quilt-example
 
-COPY --from=build /app/dist/unoffical_athena_federation_sdk-*-py3-none-any.whl /
-RUN pip install /unoffical_athena_federation_sdk-*-py3-none-any.whl
+COPY --from=build /app/dist/athena_federation-*-py3-none-any.whl /
+RUN pip install --no-cache-dir /athena_federation-*-py3-none-any.whl
 
-COPY example/ ./
-RUN ls ./
+COPY example/ ./example
```

Comment:
For some reason when I ran the `make lambda-ping` below, I got back this error:
```
Unable to import module 'example.handler': No module named 'example'
```

Looking into why, but feel free to comment/fix if you know.

Reply:
Probably a missing dependency in the Makefile for that action. I'll take a look.

Reply:
Doh! Now I'm seeing this error everywhere. I have this horrible feeling I never got it to work, but simply mis-read the error message...

Reply:
Whew.  LAMBDA_TASK_ROOT fixes it. Rolling into 0.1.3